src/components/MapboxMap/MapboxMap.vue

```diff
@@ -107,54 +112,134 @@ function initMap(accessToken: string) {
+              ['to-string', ['get', 'point_count_abbreviated']],
+            ],
+            {
+              'text-color': '#FFF',
+            },
+            '            ',
+            {},
+            ['get', 'buildingSlugModified'],
+            {
+              'text-color': '#000',
+            },
+          ],
+          'text-size': 15,
+          'text-anchor': 'left',
+          'text-justify': 'left',
+          'text-transform': 'uppercase',
+          'text-offset': [-4.25, 0],
+          'text-allow-overlap': true,
+          'text-optional': true,
+        },
+      });
+
+
+
       map.addLayer({
-        id: "points",
+        id: UNCLUSTERED_LAYER_ID,
         interactive: true,
         type: "symbol",
-        source: {
-          type: "geojson",
-          data: mapStore.geoJsonSpaces,
-          promoteId: "spaceSlug",
-        },
+        source: "clustered-points",
+        filter: ["all", ["!", ["has", "point_count"]], [">=", ["zoom"], 17]],
         layout: {
           "icon-image": [
-            "match", // A rule to determine the icon for the point...
+            "match",// A rule to determine the icon for the point...
             ["get", "buildingOccupancy"], // ... is that if the `buildingOccupancy` property matches...
             ...occupancyIconNamePairs, // ... the first element of the pair from this (flat) sequence, the second element defines the name of the icon
             "map-marker-unknown", //And the final element determines the default icon
           ],
           "icon-allow-overlap": true,
         },
       });
-
+      
       restoreMapState();
       fixInsecureLinks();
       mapStore.setMap(map);
     });
   });
-  map.on("click", (e: any) => {
+
+  map.on("click", CLUSTERS_LAYER_ID, (e) => {
+    const features = map.queryRenderedFeatures(e.point, {
+      layers: [CLUSTERS_LAYER_ID],
+    });
+
+    if (features.length) {
+      const geometry = features[0].geometry;
+
+      if (geometry.type === "Point" && geometry.coordinates) {
+        const coordinates = geometry.coordinates.slice() as [number, number];
+        map.flyTo({
+          center: coordinates,
+          zoom: 17,
+          essential: true,
+        });
+      }
+    }
+  });
+
+  // Click event handler for unclustered points
+  map.on("click", UNCLUSTERED_LAYER_ID, (e) => {
     const features = map.queryRenderedFeatures(e.point, {
-      layers: ["points"],
+      layers: [UNCLUSTERED_LAYER_ID],
     });
+
     if (features.length) {
       const properties = features[0].properties || {};
-      if (!properties.buildingSlug || !properties.spaceSlug) {
+      if (!properties.buildingSlug || !features[0].id) {
         return;
       }
       saveMapState();
+
       router.push(
         $localePath("/buildings/:buildingSlug/spaces/:spaceSlug", {
           buildingSlug: properties.buildingSlug as string,
-          spaceSlug: properties.spaceSlug as string,
+          spaceSlug: features[0].id as string,
```

Comment:
I will need help on that :) Why is this change needed?

Reply:
I can revert this back I changed it because I was doing something with a single clicked feature id with the mapbox clustering functionality itself. same as for the top ^